redaxo/src/addons/structure/update.php

```diff
@@ -21,6 +21,25 @@
     });
 }
 
+if (rex_addon::get('users')->isInstalled() && $addon->getPlugin('content')->isInstalled() && rex_version::compare($addon->getVersion(), '2.11.0-beta1', '<')) {
+    $sql = rex_sql::factory();
+    $sql->transactional(static function () use ($sql) {
+        $roles = rex_sql::factory()->setQuery('SELECT * FROM '.rex::getTable('user_role'));
+        /** @var rex_sql $role */
+        foreach ($roles as $role) {
+            $perms = $role->getArrayValue('perms');
+            $perms['options'] = $perms['options'] ?? '|';
+            $perms['options'] .= 'plublishSlice|';
```

Comment:
```suggestion
            $perms['options'] .= 'publishSlice[]|';
```

Reply:
Ggf sollten wir diese strings für die permissions als konstanten in eine klasse verlagern?

Reply:
Oder in die package.yml ?

Reply:
Ich sehe keinen vortril sie in der package.yml zu haben. Wenn es php konstanten sind, sind sie für jede IDE und psalm/phpstan erkennbar und könneen gecheckt/autocompleted werden

Reply:
(Würde ich hier im PR aber nicht mit anfangen, sondern eher separat einführen)

Reply:
> Würde ich hier im PR aber nicht mit anfangen, sondern eher separat einführen

Wollen wir es für das `structure` AddOn mal angehen? Ich würde es übernehmen.

Reply:
Gerne

Reply:
Als Entwickler braucht man die Konstanten übrigens eigentlich nie, man fragt die Rechte ja nicht selbst ab.
Aber trotzdem für uns selbst hier im Core wäre es natürlich auch nützlich.
Wobei man in der update.php wieder aufpassen muss, ob die Klasse mit den Konstanten da überhaupt geladen ist, ggf. kann man die hier gar nicht nutzen.